backend/src/db/migrations/20251021112356_add-certificate-auto-renewal.ts

```diff
@@ -0,0 +1,51 @@
+import { Knex } from "knex";
+
+import { TableName } from "../schemas";
+
+export async function up(knex: Knex): Promise<void> {
+  if (await knex.schema.hasColumn(TableName.PkiApiEnrollmentConfig, "autoRenewDays")) {
+    await knex.schema.alterTable(TableName.PkiApiEnrollmentConfig, (t) => {
+      t.renameColumn("autoRenewDays", "renewBeforeDays");
+    });
+  }
+
+  if (!(await knex.schema.hasColumn(TableName.Certificate, "renewBeforeDays"))) {
+    await knex.schema.alterTable(TableName.Certificate, (t) => {
+      t.integer("renewBeforeDays").nullable();
+      t.uuid("renewedFromCertificateId").nullable();
+      t.uuid("renewedByCertificateId").nullable();
+      t.text("renewalError").nullable();
+      t.string("keyAlgorithm").nullable();
+      t.string("signatureAlgorithm").nullable();
+      t.foreign("renewedFromCertificateId").references("id").inTable(TableName.Certificate).onDelete("SET NULL");
+      t.foreign("renewedByCertificateId").references("id").inTable(TableName.Certificate).onDelete("SET NULL");
+      t.index("renewedFromCertificateId");
+      t.index("renewedByCertificateId");
+      t.index("renewBeforeDays");
+    });
+  }
```

Comment:
IGNORE PLACEMENT OF COMMENT

A certificate that I issued through a certificate profile with no auto-renewal config shows up with renewal status of "Renews in 334d" for some reason? 

<img width="1512" height="858" alt="image" src="https://github.com/user-attachments/assets/e00ca49a-3023-4a82-8139-3f6deed90d6c" />

<img width="1500" height="456" alt="image" src="https://github.com/user-attachments/assets/e41e00e9-cc5e-4239-a9f5-bf4f00cdbdf5" />